packages/vue/src/sdk.ts

```diff
@@ -0,0 +1,414 @@
+
+// Mappings from operation to corresponding lifecycle hook.
+const HOOKS: { [key in Operation]: Hook[] } = {
+  activate: ['activated', 'deactivated'],
+  create: ['beforeCreate', 'created'],
+  destroy: ['beforeDestroy', 'destroyed'],
+  mount: ['beforeMount', 'mounted'],
+  update: ['beforeUpdate', 'updated'],
+};
+
+const COMPONENT_NAME_REGEXP = /(?:^|[-_/])(\w)/g;
+const ROOT_COMPONENT_NAME = 'root';
+const ANONYMOUS_COMPONENT_NAME = 'anonymous component';
+
+/** Vue specific configuration for Tracing Integration  */
+interface TracingOptions {
+  /**
+   * Decides whether to track components by hooking into its lifecycle methods.
+   * Can be either set to `boolean` to enable/disable tracking for all of them.
+   * Or to an array of specific component names (case-sensitive).
+   */
+  trackComponents: boolean | string[];
+  /** How long to wait until the tracked root activity is marked as finished and sent of to Sentry */
+  timeout: number;
+  /**
+   * List of hooks to keep track of during component lifecycle.
+   * Available hooks: 'activate' | 'create' | 'destroy' | 'mount' | 'update'
+   * Based on https://vuejs.org/v2/api/#Options-Lifecycle-Hooks
+   */
+  hooks: Operation[];
+}
+
+/**
+ * Inits the Vue SDK
+ */
+export function init(
+  options: Partial<Omit<VueOptions, 'tracingOptions'> & { tracingOptions: Partial<TracingOptions> }> = {},
+): void {
+  if (options.defaultIntegrations === undefined) {
+    options.defaultIntegrations = defaultIntegrations;
+  }
+  if (options.release === undefined) {
+    const window = getGlobalObject<Window>();
+    // This supports the variable that sentry-webpack-plugin injects
+    if (window.SENTRY_RELEASE && window.SENTRY_RELEASE.id) {
+      options.release = window.SENTRY_RELEASE.id;
+    }
+  }
+
+  const finalOptions = {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    Vue: getGlobalObject<any>().Vue as VueInstance,
+    attachProps: true,
+    logErrors: false,
+    tracing: false,
+    ...options,
+    tracingOptions: {
+      hooks: ['activate', 'mount', 'update'],
+      timeout: 2000,
+      trackComponents: false,
+      ...options.tracingOptions,
+    },
+  } as VueOptions;
+
+  if (finalOptions.Vue === undefined) {
+    logger.warn('No Vue instance was provided. Also there is no Vue instance on the `window` object.');
+  }
+
+  initAndBind(BrowserClient, finalOptions);
+  const client = getCurrentHub().getClient();
+  if (client) {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    client.__vueHelper = new VueHelper(finalOptions);
+  }
+
+  createVueEventProcessor();
+}
+
+/** JSDoc */
+export class VueHelper {
+  /**
+   * Cache holding already processed component names
+   */
+  private readonly _componentsCache: { [key: string]: string } = {};
+  private _rootSpan?: Span;
+  private _rootSpanTimer?: ReturnType<typeof setTimeout>;
+  private _options: Omit<VueOptions, 'Vue'> & { Vue: VueInstance };
```

Comment:
You made `VueOptions` optional, just to add it back here as required 🤔  It should be always optional.